src/main/scala/com/cognite/spark/datasource/RawTableRelation.scala

```diff
@@ -163,54 +214,70 @@ object RawTableRelation {
             .get
       ))
 
-  private def schemaWithoutRenamedKeyColumns(schema: StructType) =
+  private def schemaWithoutRenamedColumns(schema: StructType) =
     StructType.apply(schema.fields.map(field => {
       if (keyColumnPattern.findFirstIn(field.name).isDefined) {
         field.copy(name = field.name.replaceFirst("_", ""))
+      } else if (lastChangedColumnPattern.findFirstIn(field.name).isDefined) {
+        field.copy(name = field.name.replaceFirst("_", ""))
       } else {
         field
       }
     }))
 
-  private def renameKeyColumns(df: DataFrame): DataFrame = {
-    val columnsToRename = keyColumns(df.schema)
-    // rename key columns starting with the longest one first, to avoid creating a column with the same name
-    columnsToRename.sorted.foldLeft(df) { (df, keyColumn) =>
-      df.withColumnRenamed(keyColumn, s"_$keyColumn")
+  private def renameColumns(df: DataFrame): DataFrame = {
+    val columnsToRename = keyColumns(df.schema) ++ lastChangedColumns(df.schema)
+    // rename columns starting with the longest one first, to avoid creating a column with the same name
+    columnsToRename.sorted.foldLeft(df) { (df, column) =>
+      df.withColumnRenamed(column, s"_$column")
     }
   }
 
-  private def unRenameKeyColumns(df: DataFrame): DataFrame = {
-    val columnsToRename = keyColumns(df.schema)
-    // when renaming them back we instead start with the shortest key column name, for similar reasons
-    columnsToRename.sortWith(_ > _).foldLeft(df) { (df, keyColumn) =>
-      df.withColumnRenamed(keyColumn, keyColumn.substring(1))
+  private def unRenameColumns(df: DataFrame): DataFrame = {
+    val columnsToRename = keyColumns(df.schema) ++ lastChangedColumns(df.schema)
+    // when renaming them back we instead start with the shortest column name, for similar reasons
+    columnsToRename.sortWith(_ > _).foldLeft(df) { (df, column) =>
+      df.withColumnRenamed(column, column.substring(1))
     }
   }
 
   private val temporaryKeyName = s"TrE85tFQPCb2fEUZ"
+  private val temporaryLastChangedName = s"TrE85tFQPCb2fEUZnand1061"
```

Comment:
Make this something completely unique. Here you go (from `pwgen -s 16`): J2p972xzM9bf32oD